### ISSUE TITLE
Clean up the capability stack

### ISSUE THREAD

**Issue by cwgoes** [cryptiumlabs/juvix#159]
Right now, we have capabilities in a bunch of sub-modules and then `newtype`-d monads in those sub-modules to provide the requisite capabilities. I feel like this is an anti-pattern - if we want to use capabilities, we should do so throughout the codebase and really only have one monad that provides all of them when necessary (e.g. in `app/Monad.hs`), or at least a very small number of monads for common uses of the compiler pipeline.

What say you @mariari?

**Comment by mariari** [comment]
I'd be for standardizing them, however the issue that we'll face, is that a lot of these are disjoint and the effects across the codebase aren't as consistent as  say a database connection through a CRUD app.

What one probably be better is to logically group together some of the parameters in the effect system to be bundled, as to better explain their purpose instead of just being fields.

I'm curious what you'll come up with though!

**Comment by cwgoes** [comment]
> I'd be for standardizing them, however the issue that we'll face, is that a lot of these are disjoint and the effects across the codebase aren't as consistent as say a database connection through a CRUD app.
> 
> What one probably be better is to logically group together some of the parameters in the effect system to be bundled, as to better explain their purpose instead of just being fields.
> 
> I'm curious what you'll come up with though!

You are right. I think we can safely punt this until we start caring about & seriously optimising for performance, at which point we should re-evaluate effects frameworks anyways.
